### DISCUSSION TITLE
Best practice on parallelizing some light scripts across many binaries

### DISCUSSION THREAD

**Discussion by peterjmorgan** [Vector35/binaryninja-api#2039] [General] [answered]
I'm curious if you have some best practices for running some light scripts across a lot of binaries.

Right now, I'm doing this on a big R820 but if this works well I may scale it to something drastically bigger. As such, being able to take advantage of tons of RAM and lots of cores would be ideal.

**Comment by psifertex** [comment, marked as answer]
Yup! Sorry for not answering here earlier, but I've started the beginnings of some documentation about batch-scripting and some of the things to watch out for. Let me know if you've got any feedback!

https://github.com/Vector35/binaryninja-api/blob/dev/docs/guide/batch-processing.md
